src/sentry/south_migrations/0470_recent_search_backfill.py

```diff
@@ -0,0 +1,1457 @@
+# -*- coding: utf-8 -*-
+import logging
+
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import connection
+
+from sentry.utils.hashlib import md5_text
+
+
+db_query = """
+-- Make sure that we only fetch up to 30 searches per user, and filter out any
+-- that already exist
+SELECT organization_id, user_id, 0 as "type", query, date_added
+FROM (
+    SELECT user_id, query, organization_id, date_added, rank() OVER (PARTITION BY user_id ORDER BY date_added DESC) AS pos
+    FROM (
+        -- Group together all saved searches and deduplicate
+        SELECT user_id, query, organization_id, max(date_added) date_added
+        FROM (
+            (
+                -- Get all saved searches that the user explicitly set as default
+                SELECT user_id, query, sp.organization_id, max(ss.date_added) AS date_added
+                FROM sentry_savedsearch_userdefault ssud
```

Comment:
Do you need to specifically get the user defaults? Won't these same results come from the other side of the union as the users would have membership on the related project? Or is this a way to ensure that their previous default preferences end up in their new recent search data?

Reply:
That's a good point, I was just following spec but it doesn't make sense to fetch from both sources. One option could be to prioritize user defaults so that they always get included, but from my querying there are very few users with > 30 saved searches related to them, so probably not worthwhile.  I'll this section of the query